lib/widgets/post_widget.dart

```diff
@@ -1,10 +1,56 @@
+import 'dart:convert';
+
 import 'package:flutter/material.dart';
 import 'package:image_stack/image_stack.dart';
+import 'package:rice_music_sharing/screen/profile_screen.dart';
 import 'package:rice_music_sharing/widgets/post_action_button.dart';
+import '../../constants.dart';
+import 'package:http/http.dart' as http;
+import '../models/user_model.dart';
 
-class PostWidget extends StatelessWidget {
+class PostWidget extends StatefulWidget {
+  // will take in a usermodel in the future
   const PostWidget({Key? key}) : super(key: key);
 
+  @override
+  State<PostWidget> createState() => _PostWidget();
+}
+
+class _PostWidget extends State<PostWidget> {
+  late Future<UserModel> currUser;
+  String postUser = "bananalegend"; /* temporary username */
+
+  UserModel temp = UserModel(
+      netID: "jx29", displayName: "Vivian", userName: "Vivian", following: []);
+
+  void navigateToUser(BuildContext context) async {
+    currUser = fetchUser();
+    final user = await currUser;
+    // final user = temp;
+    // request user from backend
+    if (mounted) {
+      Navigator.push(
+        context,
+        MaterialPageRoute(builder: (context) => ProfileScreen(userModel: user)),
+      );
+    }
+  }
+
+  Future<UserModel> fetchUser() async {
+    final response = await http.get(Uri.parse("$backendURL\\users\\$postUser"));
```

Comment:
Is there any reason this uses `\\` instead of `/`? The latter is more typical for paths, unless I'm missing something.